sdk/sdktypes/connection.go

```diff
@@ -28,10 +28,16 @@ func (ConnectionTraits) Validate(m *ConnectionPB) error {
 }
 
 func (ConnectionTraits) StrictValidate(m *ConnectionPB) error {
+	var errProjectID, errIntegrationID error
+	if m.Name != SchedulerConnectionName {
+		errProjectID = mandatory("project_id", m.ProjectId)
+		errIntegrationID = mandatory("integration_id", m.IntegrationId)
+	}
+
 	return errors.Join(
 		mandatory("name", m.Name),
-		mandatory("project_id", m.ProjectId),
-		mandatory("integration_id", m.IntegrationId),
+		errProjectID,
+		errIntegrationID,
 	)
 }
```

Comment:
golfing

```suggestion
	var errs []error
	if m.Name != SchedulerConnectionName {
        errs = []error{
		    mandatory("project_id", m.ProjectId),
		    mandatory("integration_id", m.IntegrationId)
        }
	}

	return errors.Join(tmandatory("name", m.Name), errs...)
}
```

Reply:
you cannot mix fixed args and unpacked slices - https://stackoverflow.com/questions/28625546/mixing-exploded-slices-and-regular-parameters-in-variadic-functions
so, I did it differently than suggested